src/Sulu/Component/PHPCR/PathCleanup.php

```diff
@@ -102,7 +106,16 @@ public function cleanup($dirty, $languageCode = null)
 
         $totalParts = \count($parts);
         foreach ($parts as $i => $part) {
-            $slug = $this->slugger->slug($part, '-', $languageCode);
+            try {
+                $slug = $this->slugger->slug($part, '-', $languageCode);
+            } catch (\IntlException $e) { // handle unknown emoji transliterator id
```

Comment:
This shounds like something which should be reported to symfony itself. I think the slugger should just ignore if it does not know something.

Reply:
@alexander-schranz , this is [the condition](https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Intl/Transliterator/EmojiTransliterator.php#L43) that triggers the `IntlException`. 

It happens when there is no [mapping file](https://github.com/symfony/symfony/tree/6.2/src/Symfony/Component/Intl/Resources/data/transliterator/emoji) for a locale. For instance, there is no `emoji-de_at.php` file; which leads to [this error](https://github.com/sulu/sulu/actions/runs/3554459366/jobs/5970575754#step:10:578)

Reply:
Thank you for detailed debugging here. Let see if symfony will change here something. Created an issue here: https://github.com/symfony/symfony/issues/48364

Else I think we should just in the constructor created a `$this->sluggerWithEmoji` and `$this->sluggerWithoutEmoji` to avoid creating new objects always inside the loop.

Reply:
Done.

Reply:
Think we not longer need the sluggerWithEmoji? Since https://github.com/symfony/symfony/pull/48395?

Also I think we should do `str_replace('-', '_', $languageCode)` to convert it into a valid intl locale. @nozarashi20 What do you think?

Reply:
On other cases we did also just output the `language` this way `explode('-', $locale)[0]` but as it could be that the slugger support `en-us` and `en-uk` I would give the whole intl locale into it, if possible.

I think for general handling we should create an own issue how we maybe normalize the locale better in future. So maybe all things are handle by intl based format like `en_UK` problem is that most website don't want to have `domain.com/en_UK´ instead `domain.com/en-uk` in it. But as said that is an own issue by itself, so first concentrate on the slugger case.

Reply:
Updated code and tests